images/router/nginx/conf/nginx-config.template

```diff
@@ -5,91 +5,199 @@
+  {{ else }}
+    {{- if eq (env "ROUTER_CIPHERS" "intermediate") "old" }}
+  # Old cipher suite (maximum compatibility but insecure) from https://wiki.mozilla.org/Security/Server_Side_TLS
+  ssl_ciphers ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:ECDHE-RSA-DES-CBC3-SHA:ECDHE-ECDSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:DES-CBC3-SHA:HIGH:SEED:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!RSAPSK:!aDH:!aECDH:!EDH-DSS-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA:!SRP;
+    {{- else }}
+  # user provided list of ciphers (Colon separated list as seen above)
+  # the env default is not used here since we can't get here with empty ROUTER_CIPHERS
+  ssl_ciphers {{env "ROUTER_CIPHERS" "ECDHE-ECDSA-CHACHA20-POLY1305"}};
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+
+  server {
+    listen 1936;
+
+    location /healthz {
+      return 200 "healthy\n";
+    }
+  }
+
+  # default server
+  server {
+    listen {{env "ROUTER_SERVICE_HTTP_PORT" "80"}} default_server;
+    listen 127.0.0.1:{{env "ROUTER_SERVICE_SNI_PORT" "10444"}} default_server ssl;
 
+    server_name _;
 
+    ssl_certificate /etc/pki/tls/private/tls.crt;
+    ssl_certificate_key /etc/pki/tls/private/tls.key; 
 
-{{- range $cfgIdx, $cfg := .State }}
+    error_page 503 /error-page-503.html; 
 
+    location / {
+      return 503;
+    }
+    
+    location = /error-page-503.html {
+      root /var/lib/nginx/conf;
+    }
+  }
+
+{{- range $cfgHost, $configs := $httpAliases }}
+  {{ range $cfg := $configs }}
   upstream be_{{$cfg.Namespace}}_{{$cfg.Name}} {
-    {{- range $serviceUnitName, $weight := $cfg.ServiceUnitNames }}
-      {{- with $serviceUnit := index $.ServiceUnits $serviceUnitName }}
-        {{- range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
-    server {{$endpoint.IP}}:{{$endpoint.Port}};
+      {{ if ne $cfg.ActiveEndpoints 0 }}
+        {{- range $serviceUnitName, $weight := $cfg.ServiceUnitNames }}
+        # endpoints of {{$serviceUnitName}} 
+          {{- with $serviceUnit := index $.ServiceUnits $serviceUnitName }}
+            {{- range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
+    server {{$endpoint.IP}}:{{$endpoint.Port}} weight={{$weight}};
+            {{ end -}}
+          {{ end }}
         {{ end -}}
-      {{ end -}}
-    {{ end -}}
-  }
+      {{ else }}
+    server 127.0.0.1:{{env "ROUTER_SERVICE_UNREACHABLE_PORT" "10088"}};
```

Comment:
the dummy server is needed because NGINX cannot have an upstream without any servers ( That's why we have to insert that server when there is no active endpoints. 

When we have an upstream with the dummy server,  because there is no actual server running locally on port 10088, NGINX will return 502 bad gateway for that upstream.

I assume that the 503 code will be the correct code for the Router to return when there is no active/healthy endpoints?  In this case, I will make sure that there is a local server running on port 10088 with `location / { ... return 503; }`

Reply:
The 503 would be consistent with other templates.